js/equipe.js

```diff
@@ -0,0 +1,67 @@
+$(function() {
+
+var datasets = makeTimeSeries(window.data, window.buildDate);
+
+new Chart(document.querySelector('canvas'), {
+	type: 'line',
+	data: {
+		datasets: [{
+			data: datasets.past,
+			label: 'Membres ',
+			lineTension: 0,
+			pointRadius: 0,
+			backgroundColor: 'rgba(100, 100, 200, .6)'
+		}, {
+			data: datasets.future,
+			label: 'Présence future garantie au ' + window.buildDate.split('-').reverse().join('/'),
+			pointRadius: 0
+		}]
+	},
+	options: {
+		animation: { duration: 0 },
+		maintainAspectRatio: false,
+		scales: {
+			xAxes: [{
+				type: 'time',
+				time: { unit: 'quarter' }
+			}]
+		}
+	}
+});
+
+
+/**
+*@param	{Array<Object>}	data		An array containing objects with the following properties: `date`, an ISO-formatted date String on which an event happened, `increment`, by which Number the plotted amount has moved that day.
+*@param	{String}		splitDate	The ISO-formatted day on which the past and future datasets are split.
+*/
+function makeTimeSeries(data, splitDate) {
+	data = data.slice();  // copy to work in place
+
+	var splitPoint = {
+		date: splitDate,
+		increment: 0
+	};
+
+	data.push(splitPoint);
+
+	data.sort(function(first, second) {
+		return first.date < second.date ? -1 : 1;
+	});
+
+	var points = data.map(function(point) {
```

Comment:
Bah, c’est-à-dire que l’accumulateur va devoir wrapper à la fois le compteur et le vecteur en construction…
Sinon, on peut implémenter le compteur en allant chercher la valeur précédente à chaque fois, mais il faut alors gérer l’initialisation avec une valeur par défaut :

```js
        var points = events.reduce(function(result, event) {
		result.push({
			x: event.date,
			y: (result[result.length - 1] || { y: 0 }).y + event.increment
		});
		return result;
	}, []);
```

…même si on peut améliorer la gestion des défauts, je ne suis pas convaincu que ça soit beaucoup moins smartass que mon passage par `this` :wink:

Sinon, version plus algorithmique qui sera lisible par tous :

```js
	var currentAmount = 0;
	var points = events.map(function makeAmountPoint(event) {
		currentAmount += event.increment;
		return {
			x: event.date,
			y: currentAmount
		};
	});
```